tobiasHe/assignments/treesBFS/lc107/lc107.js

```diff
@@ -0,0 +1,79 @@
+// Problem Statement #
+
+// Given a binary tree, populate an array to represent its level-by-level traversal in reverse order, i.e., the lowest level comes first. You should populate the values of all nodes in each level from left to right in separate sub-arrays.
+
+class TreeNode {
+	constructor(value) {
+		this.value = value;
+		this.left = null;
+		this.right = null;
+	}
+}
+/* 
+  using BFS to traverse the tree, while keeping the values of each level
+  initialize a queue with root
+  keep track of queueLength, to move on to next levels
+  shift out the queue and check if currentNode.value has any left and right children
+  if there are any, add them to the queue 
+  then unshift the value of that level into levels  (because we are doing traversal in reverse order)
+  return levels 
+
```

Comment:
Don't forget to include edge cases and how you will handle them in your overview.